components/public-api/gitpod/v1/workspace.proto

```diff
@@ -72,6 +79,81 @@ message ListWorkspacesResponse {
   PaginationResponse pagination = 2;
 }
 
+message CreateAndStartWorkspaceRequest {
+
+  message Git {
+    // clone_url is the URL of the repository to clone
+    string clone_url = 1;
+
+    // ref is an alternatively symbolic. e.g. refs/tags/v1.0,
+    // empty string means the default branch of the repository
+    string ref = 2;
+
+    // create_local_branch is the branch you want to create based on provided
+    // clone_url and ref when workspace started
+    string create_local_branch = 3;
+  }
+
+  // organization_id is the ID of the organization to create the workspace
+  //
+  // +required
+  string organization_id = 1;
+
+  // configuration_id is the ID of the configuration to use
+  string configuration_id = 2;
+
+  // source describes the source refer of workspace.
+  //
+  // +required
+  oneof source {
+
+    // git describes the source refer of workspace
+    // Obtain available git using the ContextService.ParseContext operation if
+    // not sure about it.
+    Git git = 3;
+
+    // context_url is for backward compatiblity with the current dashboard, use
+    // ContextService.ParseContext get get a Git source instead
+    string context_url = 4 [ deprecated = true ];
+  }
+
+  // additional_env_variables provide additional environment variables to the
+  // workspace.
+  // It will take precedence over environment variables provided by
+  // the user and the configuration
+  repeated WorkspaceEnvironmentVariable additional_env_variables = 5;
+
+  string region = 6;
+
+  // workspace_class is the class of the workspace
+
+  string workspace_class = 7;
+
+  EditorReference editor = 8;
+
+  string name = 9;
+
+  bool pinned = 10;
+
+  // force_default_config indicates that the workspace should be created with
+  // the default configuration instead of the configuration provided in
+  // `.gitpod.yml` file
+  bool force_default_config = 11;
+}
+
+message CreateAndStartWorkspaceResponse { Workspace workspace = 1; }
+
+message StartWorkspaceRequest {
+  // workspace_id specifies the workspace that is going to start
+  //
+  // +required
+  string workspace_id = 1;
+
+  bool force_default_config = 2;
```

Comment:
@geropl @svenefftinge I'm not sure whether eventually we want to have it, or rather a new workspace should be created?

Reply:
If I somehow changed the config in my workspace so the environment doesn'T start, we should fall back to a save default configuration. I don'T know if this is something that needs to be take care of as an API argument, tbh. I'd more see it as as a fall back on start-up.

Reply:
As right now it seems a user needs to click a button on UI.

Reply:
ok, then let's keep it